client/src/LoginSignupComponents.js

```diff
@@ -0,0 +1,107 @@
+import React from "react";
+import { Button, TextField, InputLabel, Box } from "@material-ui/core";
+import { themes } from "./themes/LoginSignupThemes";
+
+function RedirectButton(props) {
+    const classes = themes();
+    return (
+        <Box className={props.className}
+            display="flex"
+            justifyContent="flex-end"
+            alignItems="baseline">
+            <span className={classes.redirectTextMargin}>{props.text}</span>
+            <Button className={classes.primary} href={props.href}>
+                {props.buttonText}
+            </Button>
+        </Box>
+    )
+}
+
+function LoginComponent(props) {
+
+    const handleOnChange = event => {
+        if (props.onChange) {
+            props.onChange(event);
+          }
+    }
+
+    const classes = themes();
+    return (
+        <Box>
```

Comment:
Look at using [Grids](https://material-ui.com/components/grid/). Grids are a wrapper for Box and have flexbox built-in. They'll make your life much easier.